misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingOptions.java

```diff
@@ -70,23 +98,91 @@ public String getWSCharacter() {
         return wsCharacter;
     }
 
+    /**
+     * @deprecated
+     * This setter method is no longer acceptable to set Formatting Options.
+     * <p> Use {@link FormattingOptions#builder()} instead.
+     */
+    @Deprecated
     public void setWSCharacter(String wsCharacter) {
         this.wsCharacter = wsCharacter;
     }
 
-    public void setColumnLimit(int columnLimit) {
-        this.columnLimit = columnLimit;
-    }
-
     public int getColumnLimit() {
         return this.columnLimit;
     }
 
+    /**
+     * @deprecated
+     * This setter method is no longer acceptable to set Formatting Options.
+     * <p> Use {@link FormattingOptions#builder()} instead.
+     */
+    @Deprecated
+    public void setColumnLimit(int columnLimit) {
+        this.columnLimit = columnLimit;
+    }
+
     public boolean getLineWrapping() {
         return lineWrapping;
     }
 
+    /**
+     * @deprecated
+     * This setter method is no longer acceptable to set Formatting Options.
+     * <p> Use {@link FormattingOptions#builder()} instead.
+     */
+    @Deprecated
     public void setLineWrapping(boolean lineWrapping) {
         this.lineWrapping = lineWrapping;
     }
+
+    public ForceFormattingOptions getForceFormattingOptions() {
+        return forceFormattingOptions;
+    }
+
+    public static FormattingOptionsBuilder builder() {
+        return new FormattingOptionsBuilder();
+    }
+
+    /**
+     * A builder for the {@code FormattingOptions}.
+     *
+     * @since 2.0.0
+     */
+    public static class FormattingOptionsBuilder {
+        private int tabSize = 4;
+        private String wsCharacter = " ";
+        private int columnLimit = 120;
+        private boolean lineWrapping = false;
+        private ForceFormattingOptions forceFormattingOptions = ForceFormattingOptions.builder().build();
+
+        public FormattingOptions.FormattingOptionsBuilder setTabSize(int value) {
+            tabSize = value;
+            return this;
+        }
+
+        public FormattingOptions.FormattingOptionsBuilder setWSCharacter(String value) {
+            wsCharacter = value;
+            return this;
+        }
+
+        public FormattingOptions.FormattingOptionsBuilder setColumnLimit(int value) {
+            columnLimit = value;
+            return this;
+        }
+
+        public FormattingOptions.FormattingOptionsBuilder setLineWrapping(boolean value) {
+            lineWrapping = value;
+            return this;
+        }
+
+        public FormattingOptions.FormattingOptionsBuilder setForceFormattingOptions(ForceFormattingOptions value) {
+            forceFormattingOptions = value;
+            return this;
+        }
```

Comment:
it'll be more readable if we can use the same field names as the parameter names.